src/lib.rs

```diff
@@ -97,18 +97,18 @@ impl AnalysisHost<CargoAnalysisLoader> {
             master_crate_map: Mutex::new(HashMap::new()),
             loader: CargoAnalysisLoader {
                 path_prefix: Mutex::new(None),
-                target: target,
+                target,
             },
         }
     }
 }
 
 impl<L: AnalysisLoader> AnalysisHost<L> {
-    pub fn new_with_loader(l: L) -> AnalysisHost<L> {
-        AnalysisHost {
+    pub fn new_with_loader(loader: L) -> Self {
```

Comment:
Is this `Self` okay here? In case of generic types I think it is less noisy and quickly allows to see that the returned generic type is the same as ours